### PR TITLE
Add jaxrs default exceptionmapper

### PR DESCRIPTION
**Pull request by jim-krueger**

### CODE REVIEW

File: dev/com.ibm.ws.jaxrs.defaultexceptionmapper/bnd.bnd [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+ javax.xml.bind.annotation, \
+ javax.xml.bind.annotation.adapters, \
+ javax.xml.bind.attachment, \
+ javax.xml.bind.helpers, \
+ javax.xml.bind.util
+ 
+Liberty-Monitoring-Components: 
+
+-dsannotations:  \
+ 	com.ibm.ws.jaxrs.defaultexceptionmapper
```
Comment (andymc12) [inline dev/com.ibm.ws.jaxrs.defaultexceptionmapper/bnd.bnd:41]: This should either explicitly add the class name or should end with ".*"
```suggestion
 	com.ibm.ws.jaxrs.defaultexceptionmapper.*
```

File: dev/com.ibm.ws.opentracing/src/com/ibm/ws/opentracing/OpentracingContainerFilter.java [hunk 2]
```diff
@@ -40,12 +43,15 @@
...
 
+import org.osgi.service.component.annotations.Component;
+
 /**
  * <p>Container filter implementation.</p>
  *
  * <p>This implementation is stateless. A single container filter is used by all applications.</p> *
  */
-public class OpentracingContainerFilter implements ContainerRequestFilter, ContainerResponseFilter, ExceptionMapper<Throwable> {
+@Component
```
Comment (andymc12) [inline dev/com.ibm.ws.opentracing/src/com/ibm/ws/opentracing/OpentracingContainerFilter.java:53]: this class will need to be added to the bnd file under `-dsannotations`

Also, it should have the same annotation attributes that I mentioned in the 1.1 project.

File: dev/com.ibm.ws.opentracing/src/com/ibm/ws/opentracing/OpentracingContainerFilter.java [hunk 3]
```diff
@@ -264,14 +270,21 @@ public void remove() {
...
+/*    @Override
     public Response toResponse(Throwable exception) {
         Tr.warning(tc, "OPENTRACING_UNHANDLED_JAXRS_EXCEPTION", exception);
         if (exception instanceof WebApplicationException) {
             return Response.fromResponse(((WebApplicationException) exception).getResponse()).header(EXCEPTION_KEY, exception).build();
         }
         return Response.serverError().header(EXCEPTION_KEY, exception).build();
     }
-
+*/
```
Comment (andymc12) [inline dev/com.ibm.ws.opentracing/src/com/ibm/ws/opentracing/OpentracingContainerFilter.java:289]: delete

File: dev/com.ibm.ws.opentracing_fat/publish/servers/opentracingFATServer1/server.xml [hunk 4]
```diff
@@ -3,6 +3,7 @@
   <featureManager>
     <feature>servlet-3.1</feature>
     <feature>opentracing-1.0</feature>
+    <feature>jaxrs-2.0</feature>
```
Comment (andymc12) [inline dev/com.ibm.ws.opentracing_fat/publish/servers/opentracingFATServer1/server.xml:6]: I'm not sure this is necessary - I thought `opentracing-1.0` already depended on `jaxrs-2.0`

**Comment by jim-krueger** [comment]
Thanks for the comments Andy.   This PR was not ready for review at the point you looked at it, but at least this way nothing was forgotten.

**Comment by jim-krueger** [comment]
#run-libby-bot

**Comment by jim-krueger** [comment]
#run-libby-bot

**Comment by jim-krueger** [comment]
#build

**Comment by jim-krueger** [comment]
#build

**Comment by jim-krueger** [comment]
#run-libby-bot

File: dev/com.ibm.ws.jaxrs.defaultexceptionmapper/src/com/ibm/ws/jaxrs/defaultexceptionmapper/DefaultExceptionMapper.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+   public Response toResponse(Throwable t) {
+     Response response;
+     if (t instanceof WebApplicationException && ((WebApplicationException)t).getResponse() != null) {
+       response = ((WebApplicationException)t).getResponse();
+     } else {
+       response = Response.serverError().build();
+     } 
+     Map<String, Object> callbackHeaders = new HashMap<>();
+     for (DefaultExceptionMapperCallback callback : this.callbacks)
+       callbackHeaders.putAll(callback.onDefaultMappedException(t, response.getStatus(), this.resourceInfo));
```
Comment (andymc12) [inline dev/com.ibm.ws.jaxrs.defaultexceptionmapper/src/com/ibm/ws/jaxrs/defaultexceptionmapper/DefaultExceptionMapper.java:54]: If we want callback implementations to be able to return null to indicate that they don't need to add any headers, then there should be a null check here to only call `putAll` if the returned map is not null.

File: dev/com.ibm.ws.jaxrs.defaultexceptionmapper/src/com/ibm/ws/jaxrs/defaultexceptionmapper/package-info.java [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+/**
+ * @version 1.0
+ */
+@org.osgi.annotation.versioning.Version("1.0")
```
Comment (andymc12) [inline dev/com.ibm.ws.jaxrs.defaultexceptionmapper/src/com/ibm/ws/jaxrs/defaultexceptionmapper/package-info.java:14]: This should include `@TraceOptions` - take a look at some of the other JAX-RS package-info.java files for examples.

**Comment by jim-krueger** [comment]
#run-libby-bot

**Comment by jim-krueger** [comment]
#build

File: dev/com.ibm.ws.jaxrs.defaultexceptionmapper/src/com/ibm/ws/jaxrs/defaultexceptionmapper/DefaultExceptionMapper.java [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+import java.util.Map;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.container.ResourceInfo;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.ExceptionMapper;
+import javax.ws.rs.ext.Provider;
+
+@Provider 
+public class DefaultExceptionMapper implements ExceptionMapper<Throwable> {
```
Comment (fmhwong) [inline dev/com.ibm.ws.jaxrs.defaultexceptionmapper/src/com/ibm/ws/jaxrs/defaultexceptionmapper/DefaultExceptionMapper.java:32]: Just nitpicking here.  The indentation of this file varies from 1-4 spaces.

File: dev/com.ibm.ws.jaxrs.defaultexceptionmapper/src/com/ibm/ws/jaxrs/defaultexceptionmapper/DefaultExceptionMapper.java [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+
+package com.ibm.ws.jaxrs.defaultexceptionmapper;
+
+import com.ibm.websphere.ras.Tr;
+import com.ibm.websphere.ras.TraceComponent;
+import com.ibm.websphere.ras.annotation.TraceObjectField;
+import com.ibm.websphere.ras.annotation.TraceOptions;
+import com.ibm.ws.ras.instrument.annotation.InjectedFFDC;import java.lang.Override;
```
Comment (fmhwong) [inline dev/com.ibm.ws.jaxrs.defaultexceptionmapper/src/com/ibm/ws/jaxrs/defaultexceptionmapper/DefaultExceptionMapper.java:18]: 2 import statements in one line

**Comment by jim-krueger** [comment]
#build

**Comment by jim-krueger** [comment]
#run-libby-bot

**Pull request merged**